configure.ac

```diff
@@ -77,6 +77,9 @@ AX_CXX_COMPILE_STDCXX([17], [noext], [mandatory])
 dnl Check if -latomic is required for <std::atomic>
 CHECK_ATOMIC
 
+dnl check if -lstdc++fs is required for std::filesystem
+CHECK_FILESYSTEM
```

Comment:
In commit "refactor: replace boost::filesystem with std::filesystem" (7810d151f5219e634b431c5393771616c759aef7)

It would be nice to split this commit up into two commits to separate the build system changes from the c++ code changes. The build system changes can be a first commit and the c++ code changes can be a second commit.